keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1004 @@
+probably just be allowed to read all TrustAnchorSets:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: trustanchorsets-reader
+rules:
+- apiGroups:
+  - certificates.k8s.io
+  resources:
+  - trustanchorsets
+  verbs:
+  - get
+  - list
+  - watch
+```
+
+### trustAnchorSets Projected Volume Source
+
+The trustAnchorSets projected volume source is responsible for retrieving a
+designated trust anchor set from a signer and publishing it into the container
+filesystem.
+
+#### Configuration Object Definition
+
+```go
+// TrustAnchorSetsProjection is the top-level configuration object for
+// instances of the trustAnchorSets kubelet projected volume source.
+type TrustAnchorSetsProjection struct {
+  // The signer to talk to.
+  SignerName string
+
+  // Which trust anchor sets should we retrieve from the signer, and where
+  // should we write them into the volume?
+  TrustAnchorSets []TrustAnchorSetFile
+}
+
+// TrustAnchorSet is a oneof that holds different strategies for writing
+// trust anchor sets to the projected volume.
+//
+// Conceptually, it’s a oneof, but only the FlatFile option is provided
+// for now.  In the future, different formats that can hold more than one
+// named trust anchor set may be supported.  Or we might support specialty
+// formats like Java keystores.
+type TrustAnchorSetFile struct {
+  // If set, write a single trust anchor set to a single PEM file.
+  FlatFile *FlatFile
+}
+
+// FlatFile is a trust anchor set provisioning strategy that writes a single
+// named trust anchor set to a single file within the project volume.
+//
+// The file format is newline-separated PEM-wrapped DER-encoded X.509
+// certificates.
+type FlatFile struct {
+  // Which named trust anchor set should we read from the signer’s
+  // TrustAnchorSets objects?
+  Name string
+  // Where should we write the flat file in the projected volume?
+  Path string
+}
+```
+
+#### Volume Content Generation and Refresh
+
+For each trustAnchorSets volume source in the pod spec:
+
+1. Kubelet will establish a watch on the TrustAnchorSets for the named signer.
+
+2. Kubelet will wait for all relevant TrustAnchorSets objects to have the
+   "Ready" condition.
+
+3. For each FlatFile entry in the volume source:
+
+   a. Kubelet will take the union of all PEMs in `spec.sets` entries with the correct
+     `name` across all TrustAnchorSets object for the signer.
+
+   b. Kubelet will write all of the resulting certificates into a file at the designated path.
+
+Any errors in this process will be surfaced as a volume mount errors that block
+the pod from starting up, since it's expected that the pod will likely crash or
+misbehave if the certificates it needs are not present.
+
+Once the pod has started, Kubelet will continue to re-evaluate the trust anchors
+sets for each file as updates arrive on its watches.  Any needed changes are
+written to the corresponding files.
+
+
+### Test Plan
```

Comment:
The KEP template Test Plan section has been [updated](https://github.com/kubernetes/enhancements/commit/a86942e8ba802d0035ec7d4a9c992f03bca7dce9).  Please pull these changes into your KEP. Thanks!

Reply:
Done